docs/grammar/BallerinaParser.g4

```diff
@@ -622,3 +623,13 @@ reservedWord
     :   FOREACH
     |   TYPE_MAP
     ;
+
+query
+    :   FROM Identifier (AS Identifier)? (JOIN Identifier (AS Identifier)? ON expression)?
```

Comment:
This grammar has multiple issues, Identifier, variableReference, and expression are used in multiple times. So it is hard to process those at parser level. Please divide each section into separate rules.

Reply:
@hasithaa Now the query is split into multiple rules. Please check